src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/Constants.cs

```diff
@@ -1,14 +1,33 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using Microsoft.Extensions.PlatformAbstractions;
+
 namespace Microsoft.AspNetCore.Server.Kestrel.Infrastructure
 {
     internal class Constants
     {
         public const int ListenBacklog = 128;
 
         public const int EOF = -4095;
-        public const int ECONNRESET = -4077;
+        public static int ECONNRESET
```

Comment:
Fetching the value would be slightly faster if you make the field a `public static readonly int` and initialize it once:

```C#
public static readonly int ECONNRESET = GetECONNRESET();

private static int GetECONNRESET() {
    switch (PlatformServices.Default.Runtime.OperatingSystemPlatform)
}
```